python/ray/serve/_private/api.py

```diff
@@ -244,6 +203,118 @@ def serve_start(
                 raise TimeoutError(
                     f"HTTP proxies not available after {HTTP_PROXY_TIMEOUT}s."
                 )
+    return controller, controller_name
+
+
+async def serve_start_async(
+    detached: bool = False,
+    http_options: Optional[Union[dict, HTTPOptions]] = None,
+    dedicated_cpu: bool = False,
+    **kwargs,
+) -> ServeControllerClient:
+    """Initialize a serve instance in async
+
+    This function is not thread-safe, caller should maintain the async lock in order
+    to start serve instance asynchronously.
+
+    This function has same functionality with ray.serve._private.api.serve_start().
+
+    Parameters & Returns are same as ray.serve._private.api.serve_start().
+    """
+
+    usage_lib.record_library_usage("serve")
+
+    try:
+        client = get_global_client(_health_check_controller=True)
+        logger.info(
+            f'Connecting to existing Serve app in namespace "{SERVE_NAMESPACE}".'
+            " New http options will not be applied."
+        )
+        if http_options:
+            _check_http_options(client, http_options)
+        return client
+    except RayServeException:
+        pass
+
+    controller, controller_name = await ray.remote(_start_controller).remote(
```

Comment:
Is there a reason `_start_controller` is a Ray task instead of an async function? I'd slightly prefer making it an async function. Making it a task will make this function show up on the Ray dashboard, which seems unnecessary.

Reply:
There are multiple blocking calls in the code path (bunch of nested blocking call), not only for controller actor starting. 
if we want to change to async task, that might refactor all the code down to the path to have an async version, i don't think it is necessary right now.

Reply:
Got it. I'm okay with keeping it as a Ray task for now. Could you

1. Add a `NOTE` or a `TODO` explaining why it's a task vs async?
2. And set `num_cpus=0` when launching the task?